torch/_refs/__init__.py

```diff
@@ -573,6 +598,12 @@ def positive(a: TensorLikeType) -> TensorLikeType:
     return a
 
 
+# real does not use _make_elementwise_unary_reference because it does not support out
+def real(a: TensorLikeType) -> TensorLikeType:
+    assert isinstance(a, TensorLike)
+    return prims.real(a)
```

Comment:
Existential prim modeling question:

One of the things we're trying to do with prims is make them orthogonal to each other. Of course we don't want to take this to extremes, but would it make more sense for `prims.real` and `prims.imag` to only work on complex tensors? Then `refs.real` would call `prims.view_of` for non-complex tensors, and only call `prims.real` on complex tensors.

Reply:
I think this makes sense. Fixed.